src/lib/crypto/kimchi_backend/gadgets/bitwise.ml

```diff
@@ -572,6 +624,66 @@ let%test_unit "bitwise rotation gadget" =
     assert (Common.is_error (fun () -> test_rot ~cs "0" 0 Left "0")) ) ;
   ()
 
+let%test_unit "bitwise shift gadgets" =
+  if tests_enabled then (
+    let (* Import the gadget test runner *)
+    open Kimchi_gadgets_test_runner in
+    (* Initialize the SRS cache. *)
+    let () =
+      try Kimchi_pasta.Vesta_based_plonk.Keypair.set_urs_info [] with _ -> ()
+    in
+
+    (* Helper to test LSL and LSR gadgets
+     *   Input operands and expected output: word len mode shifted
+     *   Returns unit if constraints are satisfied, error otherwise.
+     *)
+    let test_shift ?cs word length mode result =
+      let cs, _proof_keypair, _proof =
+        Runner.generate_and_verify_proof ?cs (fun () ->
+            let open Runner.Impl in
+            (* Set up snarky variables for inputs and output *)
+            let word =
+              exists Field.typ ~compute:(fun () ->
+                  Field.Constant.of_string word )
+            in
+            let result =
+              exists Field.typ ~compute:(fun () ->
+                  Field.Constant.of_string result )
+            in
+            (* Use the xor gate gadget *)
+            let output_shift =
+              match mode with
+              | Left ->
+                  lsl64 (module Runner.Impl) word length
+              | Right ->
+                  lsr64 (module Runner.Impl) word length
+            in
+            Field.Assert.equal output_shift result )
+      in
+      cs
+    in
+    (* Positive tests *)
+    let cs1l = test_shift "0" 1 Left "0" in
+    let cs1r = test_shift "0" 1 Right "0" in
+    let _cs = test_shift ~cs:cs1l "1" 1 Left "2" in
+    let _cs = test_shift ~cs:cs1r "1" 1 Right "0" in
+    let _cs = test_shift "256" 4 Right "16" in
+    let _cs = test_shift "256" 20 Right "0" in
+    let _cs = test_shift "6510615555426900570" 16 Right "99344109427290" in
+    (* All 1's word *)
+    let cs = test_shift "18446744073709551615" 15 Left "18446744073709518848" in
```

Comment:
Maybe make the identifier `cs` a little bit more different and more memorable (maybe `cs_all1s` than the other ignored `_cs` so that it's easier to track down to its labeled use later (`~cs:cs_all1s`).